pkg/controllers/user/helm/controller.go

```diff
@@ -103,11 +103,17 @@ func (l *Lifecycle) Updated(obj *v3.App) (runtime.Object, error) {
 		}
 		if obj.Spec.ExternalID != "" {
 			if currentRevision.Status.ExternalID == obj.Spec.ExternalID && reflect.DeepEqual(currentRevision.Status.Answers, obj.Spec.Answers) {
+				if err := l.updateForceUpgradeCondition(obj); err != nil {
```

Comment:
you don't have to update app inside controller, just set app.condition to true. The framework will update it.